test/exercises/wavelet/wavelet.chpl

```diff
@@ -0,0 +1,66 @@
+use ecgIO;
+import Math.log2;
+
+config const nLevels = 3;
+config param plotResults = false;
+
+proc main() {
+    // read sample data
+    //  (data pulled from: http://paulbourke.net/dataformats/holter/)
+    const ecgSamples = try! readEcgData("ecg_samples.txt"),
+          lenP2 = 2**(floor(log2(ecgSamples.size)):int);
+
+    // compute 1D DWT
+    const ecg_dwt = haarWavelet1D(ecgSamples[0..<lenP2], nLevels);
+
+    // generate a plot of the signal and DWT
+    if plotResults then plotDwtData(ecgSamples[0..<lenP2], ecg_dwt, nLevels);
+}
+
+// compute an n-level wavelet transform of 'x'
+proc haarWavelet1D(x: [?d] ?t, n: int) : [d] t
+    where d.rank == 1 && isNumericType(t)
+{
+    assert(2**log2(d.size) == d.size, "array size must be a power of 2");
+
+    var output : [d] t;
+    hwRec(x, output, d.size, d.size / 2**n);
+    return output;
+}
+
+// recursive helper for 'haarWavelet1D'
+proc hwRec(const signal, ref output, fmax: int, fstop: int) {
+    if fmax == fstop {
+        // store the final layer of high-pass coefficients
+        output[{0..<fmax}] = signal;
+    } else {
+        cobegin {
+            // compute and store the low-pass coefficients
+            output[{fmax/2..<fmax}] = downSample2(haarLP(signal));
+
+            // compute the high-pass coefficients and start the next layer of filtering
+            hwRec(downSample2(haarHP(signal)), output, fmax/2, fstop);
```

Comment:
I may be reading too quickly, but it seems as though this recursive expression has one task do half the work, the next task do a quarter of the work, the next task do an eighth of the work, the next a sixteenth of the work and so on, is that right?  If so, that seems like a bad load balance, and like a better approach would be to have the base case be "if I'm less than some threshold, I'll do the work here myself, and otherwise I'll make two recursive calls to `hwRec()` in a `cobegin`, similar to quicksort, mergesort, etc.

I also wonder whether there's a danger of this recursive cobegin creating too many tasks such that it should either be throttled by a serial statement once all the cores are busy, or drop into the aforementioned base-case?